backend/src/worker.rs

```diff
@@ -680,270 +1077,238 @@ print()
-    let child = Command::new("pip-compile")
+    go_path: &str,
+    preview: bool,
+) -> error::Result<String> {
+    gen_go_mymod(code, job_dir).await?;
+    let child = Command::new("go")
         .current_dir(job_dir)
-        .args(vec!["-q", "--no-header", file])
+        .args(vec!["mod", "init", "mymod"])
         .stdout(Stdio::piped())
         .stderr(Stdio::piped())
         .spawn()?;
-    *status = handle_child(job, db, logs, last_line, timeout, child).await;
-    Ok(if status.is_ok() && status.as_ref().unwrap().success() {
-        let path_lock = format!("{}/requirements.txt", job_dir);
-        let mut file = File::open(path_lock).await?;
-
-        let mut content = "".to_string();
-        file.read_to_string(&mut content).await?;
-        content = content
-            .lines()
-            .filter(|x| !x.trim_start().starts_with('#'))
-            .map(|x| x.to_string())
-            .collect::<Vec<String>>()
-            .join("\n");
-        let as_json = json!(content);
+    *status = handle_child(job_id, db, logs, last_line, timeout, child).await;
+    if status.is_ok() {
+        let child = Command::new(go_path)
+            .current_dir(job_dir)
+            .args(vec!["mod", "tidy"])
+            .stdout(Stdio::piped())
+            .stderr(Stdio::piped())
+            .spawn()?;
+        *status = handle_child(job_id, db, logs, last_line, timeout, child).await;
+    }
+    if status.is_ok() && status.as_ref().unwrap().success() {
+        if preview {
+            Ok(String::new())
+        } else {
+            let mut req_content = "".to_string();
 
-        *last_line =
-            format!(r#"{{ "success": "Successful lock file generation", "lock": {as_json} }}"#);
+            let mut file = File::open(format!("{job_dir}/go.mod")).await?;
+            file.read_to_string(&mut req_content).await?;
 
-        sqlx::query!(
-            "UPDATE script SET lock = $1 WHERE hash = $2 AND workspace_id = $3",
-            &content,
-            &job.script_hash.unwrap_or(ScriptHash(0)).0,
-            &job.workspace_id
-        )
-        .execute(db)
-        .await?;
+            req_content.push_str(&format!("\n{GO_REQ_SPLITTER}\n"));
+
+            let mut file = File::open(format!("{job_dir}/go.sum")).await?;
+            file.read_to_string(&mut req_content).await?;
+
+            Ok(req_content)
+        }
     } else {
-        sqlx::query!(
-            "UPDATE script SET lock_error_logs = $1 WHERE hash = $2 AND workspace_id = $3",
-            &logs.clone(),
-            &job.script_hash.unwrap_or(ScriptHash(0)).0,
-            &job.workspace_id
-        )
-        .execute(db)
-        .await?;
-    })
+        tracing::info!("go mod error");
+
+        Err(error::Error::ExecutionErr(format!(
+            "Lock file generation failed. Status: {status:?}",
+        )))
+    }
+}
+
+async fn gen_go_mymod(code: &str, job_dir: &String) -> error::Result<()> {
+    let code = &format!("package main\n\n{code}");
```

Comment:
This might not play well with language servers or other tooling? Those tools might expect a package statement at the start of the file but jobs can't include it because it's added here instead?

Reply:
That's true, I would have preferred using the inner separate file like we do for other languages but forcing users to declare a package name seemed more tedious than necessary